### PR TITLE
#43 Implement logged in state remember

### PR DESCRIPTION
**Pull request by sleepylee**
Closes https://github.com/sleepylee/flutter-survey/issues/43

## What happened 👀

Implement a "Remember Logged in" like feature.
 
## Insight 📝
- Add a new UseCase to check the login state at the Splash. 
- If the user has logged in before a.k.a has an access token stored, then we send the user straight to the Home page instead of Login again. Send to login otherwise.
 
## Proof Of Work 📹
First time: send to Login
Second time: send to Home

https://user-images.githubusercontent.com/13435717/119654410-a2a12e80-be52-11eb-8ad6-580a8691631d.mp4

### CODE REVIEW

**Review by markgravity — APPROVED** [review]
LGTM 🤖

File: lib/pages/splash/splash_page.dart [hunk 1]
```diff
@@ -3,25 +3,38 @@ import 'package:flutter/scheduler.dart';
...
 class SplashPage extends StatelessWidget {
   final AppNavigator _appNavigator = Get.find();
+  final splashController = Get.put(SplashController());
 
   @override
   Widget build(BuildContext context) {
     if (SchedulerBinding.instance.schedulerPhase == SchedulerPhase.idle) {
       SchedulerBinding.instance
-          .addPostFrameCallback((_) => _appNavigator.popAndNavigateToLogin());
+          .addPostFrameCallback((_) => _loginStatusCheck());
```
Comment (manh-t) [inline lib/pages/splash/splash_page.dart:16]: good to know 👍

**Review by manh-t — APPROVED** [review]

File: lib/pages/splash/splash_page.dart (hunk 1 above)
Comment (sleepylee) [reply lib/pages/splash/splash_page.dart:16]: I think normally people choose to implement the Stateful widget to get the callback after init, not sure if this way has any drawback tho. 😄

Comment (manh-t) [reply lib/pages/splash/splash_page.dart:16]: I did some research and this is one of recommend ways to handle listener without using Stateful widget.
For other state management libs, we have `ProviderListener` (Provider, Riverpod) or `BlocListener` (Bloc) widget to listen the changes but so sad I cannot find any similar things in GetX 😅

**Pull request merged**